### PR TITLE
Fixed navbar scroll spy (#798)

### PR DESCRIPTION
**Pull request by daviddavo**
### Issue
[](https://github.com/hugo-toha/toha/issues/798)

- https://github.com/hugo-toha/toha/issues/798

### Description

Followed guide in [bootstrap docs](https://getbootstrap.com/docs/4.6/components/scrollspy/#fat), and removed website url from anchors, but all browsers accept this and is common place to put just the hash-link.

### Test Evidence

Screenshot_2023-09-29_21-13-35

### CODE REVIEW

File: layouts/partials/navigators/navbar.html [hunk 1]
```diff
@@ -102,7 +102,7 @@
                   {{ if and (.section.enable) (.section.showOnNavbar) }}
                     {{ $sectionCount  = add $sectionCount 1}}
                     {{ if gt $sectionCount $maxVisibleSections }}
-                      <a class="dropdown-item" href="{{ site.BaseURL | relLangURL }}#{{ partial "helpers/get-section-id.html" . }}">{{ .section.name }}</a>
+                      <a class="dropdown-item" href="#{{ partial "helpers/get-section-id.html" . }}">{{ .section.name }}</a>
```
Comment (hossainemruz) [inline layouts/partials/navigators/navbar.html:105]: Same thing here.

File: layouts/partials/navigators/navbar.html [hunk 2]
```diff
@@ -88,7 +88,7 @@
               {{ $sectionCount  = add $sectionCount 1}}
               {{ if le $sectionCount $maxVisibleSections }}
                 <li class="nav-item">
-                  <a class="nav-link" href="{{ site.BaseURL | relLangURL }}#{{ partial "helpers/get-section-id.html" . }}">{{ .section.name }}</a>
+                  <a class="nav-link" href="#{{ partial "helpers/get-section-id.html" . }}">{{ .section.name }}</a>
```
Comment (hossainemruz) [inline layouts/partials/navigators/navbar.html:91]: This won't work on new updated navbar. We now show same menus on home and post pages. So, we can't have relative link here.

**Review by hossainemruz — COMMENTED** [review]
Thank you @daviddavo for the PR. Unfortunately, this change breaks the new updated navbar behavior.

**Comment by daviddavo** [comment]
I completely forgot about the blog feature. I think the solution would be removing the base URL just when we are on the "main" page, and keeping it on all other pages.

**Comment by hossainemruz** [comment]
>I completely forgot about the blog feature. I think the solution would be removing the base URL just when we are on the "main" page, and keeping it on all other pages.

Yes. That should work. You can use `.IsHome` to determine whether it is home page or not.
Here is an example:
https://github.com/hugo-toha/toha/blob/c943f39617c1f2b90f160df07c60fa4ec9d0be7d/layouts/partials/navigators/navbar.html#L69C63-L69C63

**Comment by daviddavo** [comment]
Should be fixed! I ended up creating a helper to avoid repeating code.

**Review by hossainemruz — APPROVED** [review]
Updated to fix home section URL

**Pull request merged**